### PR TITLE
Add github.com/confluent-kafka-go/v2/kafka instrumentation

### PR DESCRIPTION
**Pull request by eaglerayp**
Add the splunkkafkav2 package to provide support for confluent-kafka-go/kafka/v2 while maintaining compatibility.

### CODE REVIEW

File: instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafkav2/README.md
Comment (pellared) [inline instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafkav2/README.md:1]: Can you please change the directory structure from `instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafkav2` to `instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka`?

**Review by pellared — COMMENTED** [review]
Can you please sign the [Splunk Contributor License Agreement (CLA)](https://www.splunk.com/en_us/form/contributions.html)?

PS. Please let us know when you do it.

**Comment by pellared** [comment]
The GitHub workflows are failing e.g.:
- https://github.com/signalfx/splunk-otel-go/actions/runs/5250959820/jobs/9515552525?pr=2301
- https://github.com/signalfx/splunk-otel-go/actions/runs/5250959820/jobs/9515551730?pr=2301

I guess you need to update the `go.mod` and `go.sum` files.

File: instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafkav2/README.md
Comment (eaglerayp) [reply instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafkav2/README.md:1]: updated.

**Comment by eaglerayp** [comment]
updated above errors and changes.
CLA also signed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Changed
+
+- Add a new package called splunkkafkav2 to the instrumentation package of confluent-kafka-go v2.
```
Comment (pellared) [inline CHANGELOG.md:13]:
```suggestion
### Added

- Add the `github.com/signalfx/splunk-otel-go/instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka` instrumentation for the `github.com/confluentinc/confluent-kafka-go/kafka/v2` package. (#2301)
```

**Review by pellared — COMMENTED** [review]
@MrAlias PTAL. Do you think that we should support both `v1` and `v2` of [`confluent-kafka-go`](https://github.com/confluentinc/confluent-kafka-go)?

**Comment by pellared** [comment]
Build errors https://github.com/signalfx/splunk-otel-go/actions/runs/5319398623/jobs/9631849109?pr=2301:

```
Error: doc.go:15:1: package-comments: package comment should be of the form "Package splunkkafka ..." (revive)
// package splunkkafka provides functions to trace the
// github.com/confluentinc/confluent-kafka-go/kafka package.
Error: consumer.go:108:46: SA1019: c.Events is deprecated: Events (channel based consumer) is deprecated in favour of Poll(). (staticcheck)
	wrapped.events = wrapped.traceEventsChannel(c.Events())
	                                            ^
Error: producer.go:70:55: SA1019: p.ProduceChannel is deprecated: ProduceChannel (channel based producer) is deprecated in favour of Produce(). Flush() and Len() are not guaranteed to be reliable with ProduceChannel. (staticcheck)
	wrapped.produceChannel = wrapped.traceProduceChannel(p.ProduceChannel())
	                                                     ^
Error: consumer_test.go:151:3: SA1019: c.Consumer.Events is deprecated: Events (channel based consumer) is deprecated in favour of Poll(). (staticcheck)
		c.Consumer.Events() <- msg
		^
Error: consumer_test.go:[198](https://github.com/signalfx/splunk-otel-go/actions/runs/5319398623/jobs/9631849109?pr=2301#step:4:199):3: SA1019: c.Consumer.Events is deprecated: Events (channel based consumer) is deprecated in favour of Poll(). (staticcheck)
		c.Consumer.Events() <- &kafka.Message{
```

**Comment by pellared** [comment]
> In my opinion, the Splunk instrumentation library still needs to support the deprecated method for compatibility. What are your thoughts on this? @pellared

I agree with you. In order to fix the issues it would be good enough to add comments like `//nolint: staticcheck // Ensure backwards support of deprecated API.`.

**Comment by eaglerayp** [comment]
updated for build error (linter issues)

**Comment by eaglerayp** [comment]
@pellared  do we have any update on this?

**Comment by pellared** [comment]
> @pellared do we have any update on this?

I need a few days to discuss https://github.com/signalfx/splunk-otel-go/pull/2301#pullrequestreview-1487415830

File: instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Splunk instrumentation for `github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka`
+
+This instrumentation is for the
+[github.com/confluentinc/confluent-kafka-go/kafka](https://github.com/confluentinc/confluent-kafka-go)
```
Comment (pellared) [inline instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka/README.md:4]:
```suggestion
[github.com/confluentinc/confluent-kafka-go/v2/kafka (https://pkg.go.dev/github.com/confluentinc/confluent-kafka-go/v2/kafka)
```

File: instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+## Compatibility
+
+The Producer will end spans when a delivery report is returned. Setting
+`"go.delivery.reports"` to `false` will disable the delivery reports and can
+result in an build up of un-ended spans. If delivery reports are disabled, an
+un-instrumented Producer should be used instead.
+
+This instrumentation was built to support
+[v2.1.1](https://github.com/confluentinc/confluent-kafka-go/releases/tag/v2.1.1)
+of github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka. Similar to the
```
Comment (pellared) [inline instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka/README.md:16]:
```suggestion
Similar to the
```

The first sentence is redundant.

**Review by pellared — APPROVED** [review]

**Review by pellared — CHANGES_REQUESTED** [review]
I just noticed that the path was wrong in the docs.

Can you please reorganize the repo and move `github.com/signalfx/splunk-otel-go/instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka` -> `github.com/signalfx/splunk-otel-go/instrumentation/github.com/confluentinc/confluent-kafka-go/v2/kafka/splunkkafka`? Sorry that I missed it initially.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Added
+
+- Add the `github.com/signalfx/splunk-otel-go/instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/v2/splunkkafka`
+  instrumentation for the `github.com/confluentinc/confluent-kafka-go/kafka/v2`
+  package. (#2301)
```
Comment (pellared) [inline CHANGELOG.md:15]:
```suggestion
- Add the `github.com/signalfx/splunk-otel-go/instrumentation/github.com/confluentinc/confluent-kafka-go/v2/kafka/splunkkafka`
  instrumentation for the `github.com/confluentinc/confluent-kafka-go/v2/kafka`
  package. (#2301)
```

Comment (eaglerayp) [reply CHANGELOG.md]: updated

**Review by pellared — APPROVED** [review]

**Comment by eaglerayp** [comment]
Do we still miss anything?

**Pull request merged**